src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -266,48 +280,75 @@ class ObjectMonitor : public CHeapObj<mtObjectMonitor> {
 
   // Returns true if this OM has an owner, false otherwise.
   bool      has_owner() const;
-  void*     owner() const;  // Returns null if DEFLATER_MARKER is observed.
-  void*     owner_raw() const;
+  int64_t   owner() const;  // Returns NO_OWNER if DEFLATER_MARKER is observed.
+  int64_t   owner_raw() const;
+
+  // These methods return the value we set in _owner when acquiring
+  // the monitor with the given thread/vthread (tid).
+  static int64_t owner_from(JavaThread* thread);
+  static int64_t owner_from(oop vthread);
+
   // Returns true if owner field == DEFLATER_MARKER and false otherwise.
   bool      owner_is_DEFLATER_MARKER() const;
   // Returns true if 'this' is being async deflated and false otherwise.
   bool      is_being_async_deflated();
-  // Clear _owner field; current value must match old_value.
-  void      release_clear_owner(void* old_value);
+  // Clear _owner field; current value must match thread's tid.
+  void      release_clear_owner(JavaThread* thread);
   // Simply set _owner field to new_value; current value must match old_value.
-  void      set_owner_from(void* old_value, void* new_value);
-  // Simply set _owner field to current; current value must match basic_lock_p.
-  void      set_owner_from_BasicLock(void* basic_lock_p, JavaThread* current);
+  void      set_owner_from_raw(int64_t old_value, int64_t new_value);
+  // Same as above but uses tid of current as new value.
+  void      set_owner_from(int64_t old_value, JavaThread* current);
+  // Set _owner field to tid of current thread; current value must be ANONYMOUS_OWNER.
+  void      set_owner_from_BasicLock(JavaThread* current);
   // Try to set _owner field to new_value if the current value matches
   // old_value, using Atomic::cmpxchg(). Otherwise, does not change the
   // _owner field. Returns the prior value of the _owner field.
-  void*     try_set_owner_from(void* old_value, void* new_value);
+  int64_t   try_set_owner_from_raw(int64_t old_value, int64_t new_value);
+  // Same as above but uses tid of current as new_value.
+  int64_t   try_set_owner_from(int64_t old_value, JavaThread* current);
+
+  bool      has_successor();
+  bool      has_successor(JavaThread* thread);
+  void      set_successor(JavaThread* thread);
+  void      set_successor(oop vthread);
+  void      clear_successor();
+
+  // Returns true if _owner field == tid of thread, false otherwise.
+  bool has_owner(JavaThread* thread) const { return owner() == owner_from(thread); }
+  // Set _owner field to tid of thread; current value must be NO_OWNER.
+  void set_owner(JavaThread* thread) { set_owner_from(NO_OWNER, thread); }
+  // Try to set _owner field from NO_OWNER to tid of thread.
+  bool try_set_owner(JavaThread* thread) {
+    return try_set_owner_from(NO_OWNER, thread) == NO_OWNER;
+  }
 
+  bool has_owner_anonymous() const { return owner_raw() == ANONYMOUS_OWNER; }
```

Comment:
Small, drive-by comment. The rename to `has_owner_anonymous` sounds worse than the previous `is_owner_anonymous` name. I think the code reads better if you change it to `has_anonymous_owner`.

Reply:
I renamed both `set/has_owner_anonymous` to `set/has_anonymous_owner`.